src/compiler/types/lines_6500_7000.rs

```diff
@@ -5,14 +5,76 @@ use std::cell::Cell;
 use std::collections::HashMap;
 use std::rc::Rc;
 
-use super::{BaseTextRange, Node, SyntaxKind, SynthesizedComment, TextRange};
+use super::{BaseTextRange, Node, ScriptTarget, SyntaxKind, SynthesizedComment, TextRange};
 
 pub trait ModuleResolutionHost {
     fn read_file(&self, file_name: &str) -> Option<String>;
 }
 
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum Extension {
+    Ts,
+    Tsx,
+    Dts,
+    Js,
+    Jsx,
+    Json,
+    TsBuildInfo,
+    Mjs,
+    Mts,
+    Dmts,
+    Cjs,
+    Cts,
+    Dcts,
+}
+
+impl Extension {
+    pub fn maybe_from_str(extension: &str) -> Option<Self> {
```

Comment:
`Extension` gets used as a `string` in the Typescript version so expose helpers for converting back + forth to/from a `&str`